### PR TITLE
chore: add JSDoc types to @react-pdf/textkit

### PR DESCRIPTION
**Pull request by wojtekmaj**
This is a sample PR that improves type safety while keeping the code 100% in JavaScript. It's not fully functional (yet - unless #2588 gets merged), but gives an insight on how the codebase would look like with JS + JSDoc.

@diegomura, I would love to hear what you think. Should I continue?

### CODE REVIEW

**Comment by diegomura** [comment]
@wojtekmaj passing by here, but couldn't read the code yet. I'll do on the weekend! Thanks again for all the effort and ideas you're bringing

**Comment by diegomura** [comment]
@wojtekmaj needs rebase 😬

**Comment by wojtekmaj** [comment]
Just 40 files, no problem :D

File: packages/textkit/src/attributedString/advanceWidth.js [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+
 /**
  * Returns attributed string advancewidth
  *
- * @param {Object} attributedString attributed string
+ * @param {AttributedString} attributedString attributed string
  * @returns {number} advance width
  */
-const advanceWidth = (attributedString) => {
+export default function advanceWidth(attributedString) {
```
Comment (diegomura) [inline packages/textkit/src/attributedString/advanceWidth.js:13]: Nit. Any reason to switch to functions? I kind of prefer arrow fns. Not important but curious :)

**Review by diegomura — APPROVED** [review]
utACK

Left a nit question, but looks good!

File: packages/textkit/src/attributedString/advanceWidth.js (hunk 1 above)
Comment (wojtekmaj) [reply packages/textkit/src/attributedString/advanceWidth.js]: Also personal preference (and affects bundle size a bit, but that's not really relevant for react-pdf I guess). Still, regardless of motives, that is out of scope :)

**Pull request merged**